src/SpaceTime/IBackedBy.cs

```diff
@@ -0,0 +1,17 @@
+﻿
+namespace Bearded.Utilities.SpaceTime
+{
+    /// <summary>
+    /// Represents a type that is backed by a specific singular numeric type value.
+    /// </summary>
+    /// <typeparam name="T">The type of the backing value.</typeparam>
+    public interface IBackedBy<out T>
+        where T : struct
+    {
+        /// <summary>
+        /// Returns the numeric value of the type.
+        /// </summary>
+        T NumericValue { get; }
+    }
+
+}
```

Comment:
I see your points. I will see if I can think of another solution. In the worst case, I could always list all known types in awgraphics. That means that I have to keep that list up to date, but in the end things should not change too often.

In fact, there is still a chance that I may change this entire system to use doubles instead of floats. That will require some performance tests first though. In either case, your point about exposing implementation is very interesting. However, in the end there has to be some sort of property to expose the underlying value to interact with other systems.

Again, I'll consider alternatives and make some proposed changes.